moveit_core/utils/src/robot_model_test_utils.cpp

```diff
@@ -100,6 +103,32 @@ srdf::ModelSharedPtr loadSRDFModel(const std::string& robot_name)
   return srdf_model;
 }
 
+void loadIKPluginForGroup(rclcpp::Node::SharedPtr node, JointModelGroup* jmg, const std::string& base_link,
+                          const std::string& tip_link, std::string plugin, double timeout)
+{
+  using LoaderType = pluginlib::ClassLoader<kinematics::KinematicsBase>;
+  static std::weak_ptr<LoaderType> cached_loader;
+  std::shared_ptr<LoaderType> loader = cached_loader.lock();
+  if (!loader)
+  {
+    loader = std::make_shared<LoaderType>("moveit_core", "kinematics::KinematicsBase");
+    cached_loader = loader;
+  }
+
+  // translate short to long names
+  if (plugin == "KDL")
+    plugin = "kdl_kinematics_plugin/KDLKinematicsPlugin";
+
+  jmg->setSolverAllocators(
+      [=](const JointModelGroup* jmg) -> kinematics::KinematicsBasePtr {
+        kinematics::KinematicsBasePtr result = loader->createUniqueInstance(plugin);
```

Comment:
@rhaschke apparently, we have some issues finding the plugin: `C++ exception with description "According to the loaded plugin descriptions the class kdl_kinematics_plugin/KDLKinematicsPlugin with base class type kinematics::KinematicsBase does not exist. Declared types are " thrown in SetUpTestSuite().` ([link](https://github.com/ros-planning/moveit2/actions/runs/4019707767/jobs/6906859684#step:12:98497)). I also recall that there could be issues when loading plugins from the same package. I'm not sure if this was specific to ROS 2, though. Do you have any ideas?

Reply:
Looks like no plugins are found at all. No idea why. I don't have time to look into this next week.

Reply:
https://github.com/ros-planning/moveit2/pull/1901

Could this PR be related?

Reply:
unsure, it would be weird if it was related, but I would certainly be happy about that surprise. This code works just fine in MoveIt 1 and we have no issues loading IK plugins or other in a normal launch. To me it looks like a bug (or lack of proper support) to use plugins from the same package that exports them (plugin xml), not a header-not-found kind of thing.

Reply:
.. I've thought about this a bit more and I think we should move the test instance next to the IK plugin package. The test behavior really depends on the plugin implementation and we might want to test different plugins against the same behavior. Ideally, we would also test `computeCartesianPath()` with a mock implementation of an IK solver so that we are not introducing solver-specific issues into our unit tests.

Reply:
I have opened #2099 to tackle this. I have commented out the test for now so this PR can be merged